src/core/connection.c

```diff
@@ -5820,13 +5820,8 @@ QuicConnProcessRouteCompletion(
     QUIC_PATH* Path = QuicConnGetPathByID(Connection, PathId, &PathIndex);
     if (Path != NULL) {
         if (Succeeded) {
-            QuicTraceLogConnInfo(
-                SuccessfulRouteResolution,
-                Connection,
-                "Processing successful route completion Path[%hhu]",
-                PathId);
             CxPlatResolveRouteComplete(Connection, &Path->Route, PhysicalAddress, PathId);
-            QuicSendQueueFlush(&Connection->Send, REASON_ROUTE_COMPLETION);
+            (void)QuicSendFlush(&Connection->Send);
```

Comment:
I'm not sure ignoring the return is a good idea. I _think_ we need to queue the send flush if it returns indicating more needs to be sent.

Reply:
How about we instead have QuicSendFlush call QuicSendQueueFlush as appropriate? The only benefit that I can see of having QuicSendFlush return a bool is that QuicConnDrainOperations can reuse the same QUIC_OPERATION instead of freeing it and allocating a new QUIC_OPERATION from the pool.

(Note to self: we will need to also unset FlushOperationPending prior to calling QuicSendFlush in QuicConnDrainOperations if we make this change)

Reply:
I think doing the flush inline is ok, but we just need to handle the case where it can't flush everything right now.

Reply:
I'm saying instead of handling that case in the caller, we could handle it inside QuicSendFlush. So then we don't have to have the "oops, couldn't flush everything so let's queue another operation" logic in multiple places.

Reply:
I'd rather keep the operation reuse where possible.